spec/virtual-accessibility-nodes.html

```diff
@@ -0,0 +1,166 @@
+        },
+        {
+          name: "James Craig"
+          , url: "http://apple.com"
+          , mailto: "jcraig@apple.com"
+          , company: "Apple"
+          , companyURL: "http://apple.com/"
+        },
+        {
+          name: "Dominic Mazzoni"
+          , url: "http://google.com"
+          , mailto: "dmazzoni@google.com"
+          , company: "Google"
+          , companyURL: "http://google.com/"
+        },
+        {
+          name: "Alexander Surkov"
+          , url: "http://mozilla.org/"
+          , mailto: "surkov.alexander@gmail.com"
+          , company: "Mozilla"
+          , companyURL: "http://mozilla.org/"
+        },
+      ],
+      // name of the WG
+      //         wg:           "None",
+
+      // URI of the public WG page
+      //         wgURI:        "http://example.org/really-cool-wg",
+
+      // name (without the @w3c.org) of the public mailing to which comments are due
+      //          wgPublicList: "spec-writers-anonymous",
+
+      // URI of the patent status for this WG, for Rec-track documents
+      // !!!! IMPORTANT !!!!
+      // This is important for Rec-track documents, do not copy a patent URI from a random
+      // document unless you know what you're doing. If in doubt ask your friendly neighbourhood
+      // Team Contact.
+      //        wgPatentURI:  "",
+      // !!!! IMPORTANT !!!! MAKE THE ABOVE BLINK IN YOUR HEAD
+    };
+  </script>
+</head>
+
+<body>
+  <section id="introduction">
+    <h1>Introduction</h1>
+    <p>
+      This document is the spec for "Virtual Accessibility Nodes", a part of the AOM project. For background, please refer to the
+      <a href="https://github.com/WICG/aom/blob/master/explainer.md">Accessibility Object Model Explainer</a> for the background and motivation.
+    </p>
+  </section>
+  <section class="virtual">
+    <h2>
+      Virtual Accessibility Nodes
+    </h2>
+    <p>
+      Virtual Accessibility Nodes will allow authors to expose "virtual" accessibility nodes, which are not associated directly
+      with any particular DOM node, to assistive technology.
+    </p>
+    <p>
+      This mechanism is often present in native accessibility APIs, in order to allow authors more granular control over the accessibility
+      of custom-drawn APIs.
+    </p>
+    <p>
+      On the web, this would allow:
+      <ul>
+        <li>creating an accessible alternative to canvas-based UI which does not rely on fallback or visually-hidden DOM content.</li>
+        <li>adding semantic information to other types of non-DOM based UI, such as a remote desktop implementation using a
+          <code>&lt;video&gt;</code> element, or another novel form of UI presentation.</li>
+      </ul>
+    </p>
+  </section>
+  <section>
+    <!--
+Look at:
+- https://html.spec.whatwg.org/multipage/media.html#texttrack
+- https://developer.mozilla.org/en-US/docs/Web/API/AudioNode
+
+Do we need to have three+ AccessibleNode types?
```

Comment:
I think we need AccessibleNode and AccessibleRoot, if we want to have the semantics where the AccessibleRoot is just a container for the children and doesn't have its own semantics.

I don't understand why we need both AccessibleNode and VirtualAccessibleNode. I thought we got rid of the use cases for a non-virtual AccessibleNode.

Reply:
AccessibleNode is the common ancestor of VirtualAccessibleNode and AccessibleRoot.